hw_3/task_1.py

```diff
@@ -0,0 +1,42 @@
+'''
+Создать форму для регистрации пользователей на сайте.
+Форма должна содержать поля "Имя", "Фамилия", "Email", "Пароль" и кнопку "Зарегистрироваться".
+При отправке формы данные должны сохраняться в базе данных, а пароль должен быть зашифрован.
+'''
+
+from flask import Flask, redirect, url_for, render_template
+from flask_wtf.csrf import CSRFProtect
+from hw_3.form_1 import RegistrationForm
+from hw_3.models_1 import db, User
+from hashlib import sha256
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///mydatabase.db'
+db.init_app(app)
+app.config['SECRET_KEY'] = b'aa07dcc1c9c38e29df1bc19e14a764c18f9ce000a1a6fc9e385aabe6bbd02f1d'
+csrf = CSRFProtect(app)
+
+
+@app.route('/', methods=['GET', 'POST'])
+def index():
+    form = RegistrationForm()
+    if form.validate_on_submit():
+        user = User(first_name=form.first_name.data,
+                    last_name=form.last_name.data,
+                    email=form.email.data,
+                    password=sha256(form.password.data.encode(encoding='utf-8')).hexdigest(),
+                    confirm_password= sha256(form.password.data.encode(encoding='utf-8')).hexdigest())
+
+        db.session.add(user)
+        db.session.commit()
+        return redirect(url_for('index'))
+    return render_template('index.html', form=form)
```

Comment:
У нас нет ни одного сообщения в случае, если форма не прошла валидацию, как и в случае, когда данные записываются в базу данных. Тут можно, например применить функцию `flash`
```suggestion
        flash('Регистрация прошла успешно')
        return redirect(url_for('index'))
    else:
        for field in form:
            for error in field.errors:
                flash(error)
    return render_template('index.html', form=form)
```